src/image-copy/azext_imagecopy/custom.py

```diff
@@ -0,0 +1,135 @@
+    logger.warn("Getting os disk id of the source vm/image")
+    cmd = prepare_cli_command([source_type, 'show', \
+        '--name', source_object_name, \
+        '--resource-group', source_resource_group_name])
+
+    json_cmd_output = run_cli_command(cmd, True)
+
+    source_os_disk_id = json_cmd_output['storageProfile']['osDisk']['managedDisk']['id']
+    source_os_type = json_cmd_output['storageProfile']['osDisk']['osType']
+    logger.debug("source_os_disk_id: " + source_os_disk_id + " source_os_type: " + source_os_type)
+
+
+    # create source snapshots
+    logger.warn("Creating source snapshot")
+    source_os_disk_snapshot_name = source_object_name + '_os_disk_snapshot'
+    cmd = prepare_cli_command(['snapshot', 'create', \
+        '--name', source_os_disk_snapshot_name, \
+        '--resource-group', source_resource_group_name, \
+        '--source', source_os_disk_id])
+
+    run_cli_command(cmd)
+
+
+    # Get SAS URL for the snapshotName
+    logger.warn("Getting sas url for the source snapshot")
+    cmd = prepare_cli_command(['snapshot', 'grant-access', \
+        '--name', source_os_disk_snapshot_name, \
+        '--resource-group', source_resource_group_name, \
+        '--duration-in-seconds', '3600'])
+
+    json_output = run_cli_command(cmd, True)
+
+    source_os_disk_snapshot_url = json_output['accessSas']
+    logger.debug(source_os_disk_snapshot_url)
+
+
+    # Start processing in the target locations
+
+    transient_resource_group_name = 'image-copy-rg'
+    create_resource_group(transient_resource_group_name, 'eastus')
+
+    target_locations_list = target_location.split(',')
+    target_locations_count = len(target_locations_list)
+    logger.warn("Target location count: {0}".format(target_locations_count))
+
+    create_resource_group(target_resource_group_name, target_locations_list[0].strip())
+
+    if parallel_degree == -1:
+        pool = Pool(target_locations_count)
+    elif parallel_degree == 1:
+        pool = Pool(1)
+    else:
+        pool = Pool(min(parallel_degree, target_locations_count))
+
+    # try to get a handle on arm 409s
+    azure_pool_frequency = 5
+    if target_locations_count >= 5:
+        azure_pool_frequency = 15
+    elif  target_locations_count >= 3:
+        azure_pool_frequency = 10
+
+    tasks = []
+    for location in target_location.split(','):
+        location = location.strip()
+        tasks.append((location, transient_resource_group_name, source_type, \
+        source_object_name, source_os_disk_snapshot_name, source_os_disk_snapshot_url, \
+        source_os_type, target_resource_group_name, azure_pool_frequency))
+
+    logger.warn("Starting async process for all locations")
+
+    for task in tasks:
+        pool.apply_async(create_target_image, task)
+
+    try:
+        pool.close()
+        pool.join()
+    except KeyboardInterrupt:
+        print('xxx - parent')
+        logger.warn('User cancelled the operation')
+        if 'true' in cleanup:
```

Comment:
If `cleanup` is defined as `action='store_true'` then this would simply be `if cleanup` with is a lot less likely to break e.g. with letter casing.